### ISSUE TITLE
Making link to add-on listing page accessible before it passes review

### ISSUE THREAD

**Issue by brampitoyo** [mozilla/addons#39]
On [#33](https://github.com/mozilla/addons/pull/33#issuecomment-184900219), @jvillalobos wrote.

>  When an add-on is awaiting its first review, we allow the add-on page to be accessible to people who have the URL […]

If we make the link to my add-on page accessible, even when my add-on hasn’t passed review, what’s stopping me from publishing my link on my website or social media and telling others to download it?

But I assume that we take this risk for a very good reason: we want add-on developers to be able to get add-ons in the hands of testers as soon as possible. Do we have any other reason?

My proposal: instead of making the release channel more “insecure” by giving access to anybody with the link, why not relax the beta channel and keep release channel closed until review is successful?

In detail:
- If version channel is listed release, listing page is accessible only after add-on status is approved
- If version channel is listed beta, add-on listing page is accessible while add-on status is awaiting review

But maybe this is complicating things. What do you think?

**Comment by brampitoyo** [comment]
@andymckay Is it helpful if we create a “Design” or “UX” label for these kinds of issues? I would love to move my email threads to GitHub issues, where it’s more focused.

**Comment by jvillalobos** [comment]
> If we make the link to my add-on page accessible, even when my add-on hasn’t passed review, what’s stopping me from publishing my link on my website or social media and telling others to download it?
Nothing, but we have warnings to deter people from downloading. Now that we require signing, it's even less likely that someone can successfully distribute such a submission. Also, this is only temporary while the add-on awaits review, which in the majority of cases should take a couple of weeks at most.
> Do we have any other reason?
No, allowing developers to share the add-ons with testers is the main reason.
> why not relax the beta channel and keep release channel closed until review is successful?
We haven't had any problems with this feature so far, so I'm not concerned from a security standpoint. Especially not now, given that the files are unsigned and Firefox will generally refuse to install them. So, I think the complexity of making that distinction isn't really worthwhile.

**Comment by andymckay** [comment]
Go for it, added a UX label for ya.

I would just add that the visibility of an add-on status is a) complicated and b) unclear. I think we can do better than buttons striped in different colours for example.

**Comment by pwalm** [comment]
@bram Was this issue solved with the redesign of the devhub site?

**Comment by brampitoyo** [comment]
@pwalm I don’t think we’ve changed our URL sharing policy, but @jvillalobos has refactored our add-on statuses and made them a lot simpler. This clarified a lot of things that you can and cannot do under specific status. I think the problem has been largely addressed, so let’s close this issue.
